docs/aca/06-aca-dapr-bindingsapi/TasksNotifierController.cs

```diff
@@ -0,0 +1,79 @@
+using Dapr.Client;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using TasksTracker.Processor.Backend.Svc.Models;
+
+namespace TasksTracker.Processor.Backend.Svc.Controllers
+{
+    [Route("api/tasksnotifier")]
+    [ApiController]
+    public class TasksNotifierController : ControllerBase
+    {
+        private readonly IConfiguration _config;
+        private readonly ILogger _logger;
+        private readonly DaprClient _daprClient;
+        public TasksNotifierController(IConfiguration config, ILogger<TasksNotifierController> logger, DaprClient daprClient)
+        {
+            _config = config;
+            _logger = logger;
+            _daprClient = daprClient;
+        }
+
+        [Dapr.Topic("dapr-pubsub-servicebus", "tasksavedtopic")]
+        [HttpPost("tasksaved")]
+        public async Task<IActionResult> TaskSaved([FromBody] TaskModel taskModel)
+        {
+            _logger.LogInformation("Started processing message with Task Name '{0}'", taskModel.TaskName);
+
+            var sendGridResponse = await SendEmail(taskModel);
+
+            if (sendGridResponse)
+            {
+                return Ok();
+            }
+
+            return BadRequest("Failed to send an email");
+        }
+
+        private async Task<bool> SendEmail(TaskModel taskModel)
+        {
+            var integrationEnabled = _config.GetValue<bool>("SendGrid:IntegrationEnabled");
+            var sendEmailResponse = true;
+            var subject = $"Task '{taskModel.TaskName}' is assigned to you!";
+            var plainTextContent = $"Task '{taskModel.TaskName}' is assigned to you. Task should be completed by the end of: {taskModel.TaskDueDate.ToString("dd/MM/yyyy")}";
+
+            try
+            {
+                //Send actual email using Dapr SendGrid Outbound Binding (Disabled when running load test)
+                if (integrationEnabled)
+                {
+                    IReadOnlyDictionary<string, string> metaData = new Dictionary<string, string>()
+                {
+                    { "emailTo", taskModel.TaskAssignedTo },
+                    { "emailToName", taskModel.TaskAssignedTo },
+                    { "subject", subject }
+                };
+                    await _daprClient.InvokeBindingAsync("sendgrid", "create", plainTextContent, metaData);
+                }
+                else
+                {
+                    //Introduce artificial delay to slow down message processing
+                    _logger.LogInformation("Simulate slow processing for email sending for Email with Email subject '{0}' Email to: '{1}'", subject, taskModel.TaskAssignedTo);
+                    Thread.Sleep(5000);
```

Comment:
maybe delay for 1000 milliseconds is enough to see scale out and scale in within reasonable time.